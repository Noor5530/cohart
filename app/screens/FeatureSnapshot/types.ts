import { UserState } from "models/reducers/user";
export type Post = {
  _id: string | number;
  user_id: string | number;
  created_at: string;
  updated_at: string;
  title: string;
  description: string;
  tags: string;
  processing_status: string;
  post_image: string;
  post_file: string;
  jw_media_id: string;
  jw_media_url: string;
  jw_media_thumb: string;
  post_type: string;
  is_hide: boolean;
  created_by: UserState;
  glow_count: number;
  is_featured: boolean;
};

type user = {};

export interface FeatureSnapshotProps {
  item: Post;
}
