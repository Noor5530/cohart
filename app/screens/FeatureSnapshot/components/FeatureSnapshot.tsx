import React from "react";
import { useNavigation } from "@react-navigation/native";
import { FeatureSnapshotProps } from "../types";
import { widthPercentageToDP } from "react-native-responsive-screen";
import AnimatedImage from "components/AnimatedImage";
import { TouchableOpacity, StyleSheet } from "react-native";
import { shimmerColor } from "../../../config/colors";
const FeatureSnapshotList = (props: FeatureSnapshotProps) => {
  const { item } = props;
  const navigation = useNavigation();
  const onPress = () => {
    navigation.navigate("NewPostPreview", {
      post: item,
      back: true,
      from: "featureSnapshot",
    });
  };
  return (
    <TouchableOpacity
      testID="featureSnapShots"
      onPress={onPress}
      style={styles.container}
    >
      <AnimatedImage
        thumbnail={item?.jw_media_thumb}
        style={styles.image}
        resizeMode="cover"
        uri={item?.jw_media_thumb}
      />
    </TouchableOpacity>
  );
};

export default FeatureSnapshotList;

const styles = StyleSheet.create({
  container: {
    height: 160,
    width: widthPercentageToDP(31),
    backgroundColor: "white",
  },
  image: { flex: 1, margin: 5, backgroundColor: shimmerColor },
});
