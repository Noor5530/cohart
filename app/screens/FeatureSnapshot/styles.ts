import { appFonts } from "components/text";
import { StyleSheet } from "react-native";
const styles = StyleSheet.create({
  container: { backgroundColor: "white", flex: 1 },
  listContainer: {
    alignItems: "center",
    backgroundColor: "white",
  },
  listComponent: {
    alignSelf: "center",
    alignItems: "center",
    justifyContent: "center",
    paddingTop: 20,
  },
  emptyText: {
    fontSize: 14,
    fontFamily: appFonts.InterBlack,
  },
});
export default styles;
