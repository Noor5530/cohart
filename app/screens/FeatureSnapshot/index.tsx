import React, { useEffect, useState, useCallback } from "react";

import { useDispatch } from "react-redux";
import { getFeaturedPosts } from "services/getFeaturedPosts";
import { enableSnackBar } from "../../store/actions/snackBarAction";
import { useIsFocused } from "@react-navigation/native";
import FeatureSnapshot from "./components/FeatureSnapshot";
import { FlatList, View, Text } from "react-native";
import Header from "components/Header";
import styles from "./styles";
import { Post } from "./types";
const FeatureSnapshotList = () => {
  const [featureSnapShort, setFeatureSnapShort] = useState([]);
  const [loading, setLoading] = useState(false);
  const isFocused = useIsFocused();
  const dispatch = useDispatch();
  useEffect(() => {
    if (isFocused) {
      fetchFeatureSnapShort();
    }
  }, [isFocused]);
  const fetchFeatureSnapShort = useCallback(async () => {
    try {
      setLoading(true);
      let response = await getFeaturedPosts({ admin_view: true });
      setFeatureSnapShort(response?.data?.data);
      setLoading(false);
    } catch (error) {
      console.log(error);
      dispatch(enableSnackBar());
      setLoading(false);
    }
  }, [setLoading, setFeatureSnapShort]);
  const renderItem = ({ item }: { item: Post }) => {
    return <FeatureSnapshot item={item} />;
  };
  const ListEmptyComponent = () => {
    return (
      <View style={styles.listComponent}>
        {loading ? null : <Text style={styles.emptyText}>No post found</Text>}
      </View>
    );
  };
  return (
    <View style={styles.container}>
      <Header back />
      <FlatList
        testID={"SnapShotList"}
        onRefresh={fetchFeatureSnapShort}
        refreshing={loading}
        style={styles.container}
        contentContainerStyle={styles.listContainer}
        ListEmptyComponent={ListEmptyComponent}
        keyExtractor={(item, index) => `id_${index}`}
        data={featureSnapShort}
        numColumns={3}
        renderItem={renderItem}
      />
    </View>
  );
};

export default FeatureSnapshotList;
