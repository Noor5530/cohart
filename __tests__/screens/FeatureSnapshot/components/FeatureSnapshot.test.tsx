import "react-native";
import React from "react";
import FeatureSnapshot from "../../../../app/screens/FeatureSnapshot/components/FeatureSnapshot";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react-native";

jest.useFakeTimers();

test("renders correctly", () => {
  const tree = renderer.create(<FeatureSnapshot item={null} />).toJSON();

  expect(tree).toMatchSnapshot();
});
test("renders correctly", () => {
  const { getByTestId } = render(<FeatureSnapshot item={null} />);

  fireEvent.press(getByTestId("featureSnapShots"));
  expect(getByTestId("featureSnapShots")).not.toBeNull();
});
