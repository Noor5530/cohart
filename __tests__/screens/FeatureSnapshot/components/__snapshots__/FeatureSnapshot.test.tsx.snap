// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "height": 160,
      "opacity": 1,
      "width": undefined,
    }
  }
  testID="featureSnapShots"
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "alignSelf": "center",
            "bottom": 0,
            "position": "absolute",
            "top": 0,
            "zIndex": 4,
          },
          Object {
            "backgroundColor": "#e1e4e8",
            "opacity": 1,
          },
          Object {
            "backgroundColor": "#E7E5E7",
            "flex": 1,
            "margin": 5,
          },
        ],
      ]
    }
  >
    <FastImageView
      blurRadius={3}
      resizeMode="cover"
      source={
        Object {
          "priority": "low",
          "uri": undefined,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "backgroundColor": "#E7E5E7",
            "flex": 1,
            "margin": 5,
          },
        ],
      ]
    }
  >
    <FastImageView
      onFastImageError={[Function]}
      onFastImageLoad={[Function]}
      resizeMode="cover"
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
